app/src/main/AndroidManifest.xml

```diff
@@ -61,20 +61,32 @@
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
         </activity>
+        <service android:name="org.jboss.aerogear.android.unifiedpush.gcm.UnifiedPushInstanceIDListenerService" android:exported="false">
+            <intent-filter>
+                <action android:name="com.google.android.gms.iid.InstanceID"/>
+            </intent-filter>
+        </service>
+
 
         <receiver
+            android:name="com.google.android.gms.gcm.GcmReceiver"
+            android:exported="true"
+            android:permission="com.google.android.c2dm.permission.SEND" >
+            <intent-filter>
+                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
+                <category android:name="org.feedhenry.pushstarter"/>
+            </intent-filter>
+        </receiver>
+        <service
             android:name="org.jboss.aerogear.android.unifiedpush.gcm.AeroGearGCMMessageReceiver"
             android:permission="com.google.android.c2dm.permission.SEND">
             <intent-filter>
                 <action android:name="com.google.android.c2dm.intent.RECEIVE" />
-                <action android:name="com.google.android.c2dm.intent.REGISTRATION" />
-
-                <category android:name="org.feedhenry.pushstarter" />
             </intent-filter>
             <meta-data
                 android:name="DEFAULT_MESSAGE_HANDLER_KEY"
                 android:value="org.jboss.aerogear.unifiedpush.helloworld.handler.NotificationBarMessageHandler" />
```

Comment:
@secondsun can we change to to... some random value? e.g.

```
org.feedhenry.pushstarter..handler.NotificationBarMessageHandler
```

Or does it have to match something ?